python/lsst/sims/alertsim/alertsim_main.py

```diff
@@ -132,9 +138,11 @@ def get_sender(protocol, ipaddr, port, header):
     @param [out] is object of the broadcast child class
     
     """
-    return vars(broadcast.broadcast)[protocol](ipaddr, port, header)
+    return vars(broadcast)[protocol](ipaddr, port, header)
 
-def iter_and_serialize(obs_data, obs_metadata, observations_field, history):
+def query_and_dispatch(obs_data, obs_metadata, observations_field, 
+                       history, session_dir, sender, radius, opsim_path, 
+                       full_constraint, serialize_json):
```

Comment:
I think you need to update the docstring of `query_and_dispatch`.  You say that `obs_data` is the data from a catsim query, but at line 199, you treat it like an `InstanceCatalog` instance (i.e. you call `obs_data.iter_catalog()`, which makes me think `obs_data` is an instantiation of an `InstanceCatalog` class).

Reply:
You're right.